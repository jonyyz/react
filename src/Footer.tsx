import * as React from "react";

import {
  Theme,
  makeStyles,
  createStyles
} from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";

const useStyles = makeStyles(({ palette, spacing }: Theme) => createStyles({
  footer: {
    padding: spacing(2),
    marginTop: "auto",
    backgroundColor: palette.primary.main,
    color: palette.getContrastText(palette.primary.main)
  }
}));

const Footer: React.FunctionComponent = () => {
  const classes = useStyles({});

  return (
    <div className={classes.footer}>
      <Typography variant="body1">
        <Link
          href="https://www.carbonite.com/"
          target="_blank"
          rel="noopener noreferrer"
          color="inherit"
        >
          Visit Carbonite!
        </Link>
      </Typography>
    </div>
  );
}

export default Footer;
