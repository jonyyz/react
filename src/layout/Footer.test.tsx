import * as React from "react";
import { shallow } from "enzyme";

import Footer from "./Footer";

describe("Footer", () => {
  test("should render", () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
});
