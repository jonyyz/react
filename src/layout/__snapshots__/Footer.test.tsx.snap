// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render 1`] = `
<div
  className="makeStyles-footer-1"
>
  <WithStyles(ForwardRef(Typography))
    variant="body1"
  >
    <WithStyles(ForwardRef(Link))
      color="inherit"
      href="https://www.carbonite.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Visit Carbonite!
    </WithStyles(ForwardRef(Link))>
  </WithStyles(ForwardRef(Typography))>
</div>
`;
